### PR TITLE
52 auth frontend

### PR DESCRIPTION
**Pull request by SaifHayek**

### CODE REVIEW

File: client/src/hooks/useAuth.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  signOut: () => {},
+}
+
+export const AuthContext = createContext<IAuthContext>(initContext)
+
+export const useProvideAuth = ():IAuthContext => {
+  const [user, setUser] = useState<IUser | null>(null)
+  const signIn = async (payload:object): Promise<object> => {
+    try {
+      const userInfo = await ApiService.post('/api/v1/users/login', payload)
```
Comment (aaamra) [inline client/src/hooks/useAuth.tsx:28]: i think if we added the `/api/v1` in the base url my be better
```suggestion
      const userInfo = await ApiService.post('/users/login', payload)
```

File: client/src/hooks/useAuth.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      }
+      return { isLogged: true }
+    } catch (err:any) {
+      setUser(null)
+      return { isLogged: false, error: err }
+    }
+  }
+  const signUp = async (payload:object):Promise<object> => {
+    try {
+      const userInfo = await ApiService.post('/api/v1/users/signup', payload)
```
Comment (aaamra) [inline client/src/hooks/useAuth.tsx:41]: same here
```suggestion
      const userInfo = await ApiService.post('/users/signup', payload)
```

File: client/src/hooks/useAuth.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  }
+  const signOut = ():void => {
+    JwtService.destroyToken()
+    setUser(null)
+  }
+
+  useEffect(() => {
+    const getUser = async ():Promise<void> => {
+      try {
+        const userInfo = await ApiService.get('/api/v1/users/me')
```
Comment (aaamra) [inline client/src/hooks/useAuth.tsx:60]:
```suggestion
        const userInfo = await ApiService.get('/users/me')
```

**Review by aaamra — COMMENTED** [review]
great work @SaifHayek 

but i have small comments

**Review by shamskhodary — APPROVED** [review]

**Review by Bakeza — APPROVED** [review]

**Review by MostafaBalousha123 — APPROVED** [review]

**Review by aaamra — APPROVED** [review]

**Pull request merged**